grammar.js

```diff
@@ -840,6 +840,7 @@ module.exports = grammar(C, {
 
     operator_name: $ => token(seq(
       'operator',
+      /\s*/,
```

Comment:
Since a space is allowed, did you try parsing `operator_name` as an internal rule, containing separate tokens for the `operator` keyword and the operator symbol? I believe it would just entail removing the `token(` wrapper. It may create other problems, but I think it might be a better solution if it works.

Reply:
Before I rebased my PR, it was working without `token` but after rebasing on top of https://github.com/tree-sitter/tree-sitter-cpp/commit/95703675283618e8860f4de348f05e038bf912e8 `token` is required else the tests for cast operator are failing.

Reply:
It's pretty strange: when generating there are no conflicts and when parsing:
```cpp
operator int() const;
```
I get 
```lisp
(translation_unit [0, 0] - [1, 0]
  (ERROR [0, 0] - [0, 20]
    (primitive_type [0, 9] - [0, 12]))
  (expression_statement [0, 20] - [0, 21]))
```